src/lib/external_lgpl/tecs/tecs.cpp

```diff
@@ -447,29 +447,35 @@ void TECS::_update_pitch(void)
 	_SEB_error = SEB_dem - (_SPE_est * SPE_weighting - _SKE_est * SKE_weighting);
 	_SEBdot_error = SEBdot_dem - (_SPEdot * SPE_weighting - _SKEdot * SKE_weighting);
 
+	// Calculate factor relating an error in specific energy to a desired delta pitch angle
+	float gainInv = _integ5_state * _timeConst * CONSTANTS_ONE_G;
+
 	// Calculate integrator state, constraining input if pitch limits are exceeded
 	float integ7_input = _SEB_error * _integGain;
 
 	if (_pitch_dem_unc > _PITCHmaxf) {
-		integ7_input = min(integ7_input, _PITCHmaxf - _pitch_dem_unc);
+		integ7_input = min(integ7_input, 0.0f);
+		// integ7_input = min(integ7_input, (_PITCHmaxf - _pitch_dem_unc) * gainInv / _DT);
```

Comment:
This piece of code appear will prevent the integrator from winding up in a direction that will increase the saturation, however it will not reduce the integrator to reduce the amount of saturation, which was what the code on the commented out line 458 did. I think a hybrid of the two is a better solution whereby the integrator cannot wind forward into saturation, but will decay back to a value that removes the saturation at a time constant that is explicitly defined.